components/dashboard/src/components/ConfirmationModal.tsx

```diff
@@ -35,6 +35,7 @@ export default function ConfirmationModal(props: {
     const buttonDisabled = useRef(props.buttonDisabled);
     useEffect(() => {
         buttonDisabled.current = props.buttonDisabled;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What's the reason the linter gets this one wrong so often? And if it's the case, can we tweak the rules/disable?

Reply:
I think it's more often use overly-using or improperly using the `useEffect()` hook (from what I've bumped into so far). This linter rule makes sure you include any "dependencies" for your hook, cause if you miss some, it can behave unexpectedly (using stale variable values), so I think it's still preferred to keep the hook as is, and when we have an exception, it's called out with a disabled line like this. In many of these fixes I've opted for the safer approach of not changing existing behavior and disabling the warning vs. adjusting behavior to what might be more correct.